server/src/main/java/org/elasticsearch/common/geo/EdgeTreeReader.java

```diff
@@ -33,10 +33,12 @@
 public class EdgeTreeReader implements ShapeTreeReader {
     final ByteBufferStreamInput input;
     final int startPosition;
+    private final boolean closed;
 
-    public EdgeTreeReader(ByteBufferStreamInput input) throws IOException {
+    public EdgeTreeReader(ByteBufferStreamInput input, boolean closed) throws IOException {
```

Comment:
Could you add a comment about meaning of closed here?

Reply:
will do! I'll update to clarify

A edge-tree representing a shape is open when the edges do not all connect at the beginning and end. it is closed when the edges do connect at the beginning and the end.